pkg/types/interfacetests/chain_components_interface_tests.go

```diff
@@ -52,22 +44,26 @@ var AnySliceToReadWithoutAnArgument = []uint64{3, 4}
 
 const AnyExtraValue = 3
 
-func RunChainReaderInterfaceTests[T TestingT[T]](t T, tester ChainReaderInterfaceTester[T]) {
-	t.Run("GetLatestValue for "+tester.Name(), func(t T) { runChainReaderGetLatestValueInterfaceTests(t, tester) })
-	t.Run("BatchGetLatestValues for "+tester.Name(), func(t T) { runChainReaderBatchGetLatestValuesInterfaceTests(t, tester) })
+func RunChainComponentsInterfaceTests[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], mockRun bool) {
+	t.Run("GetLatestValue for "+tester.Name(), func(t T) { runChainComponentsGetLatestValueInterfaceTests(t, tester, mockRun) })
+	t.Run("BatchGetLatestValues for "+tester.Name(), func(t T) { runChainComponentsBatchGetLatestValuesInterfaceTests(t, tester, mockRun) })
 	t.Run("QueryKey for "+tester.Name(), func(t T) { runQueryKeyInterfaceTests(t, tester) })
 }
```

Comment:
These test cases are all for ContractReader, so the naming should be:
runChainComponentsGetLatestValueInterfaceTests -> `runContractReaderGetLatestValueInterfaceTests`.

I would keep also keep the RunContractReaderInterfaceTests and add same for ChainWriter later, its better to keep this granular, especially because in the relayers we will test these two components in separate places.

Reply:
@ilija42 I disagree in this case actually. Most of these tests are inherently testing the ChainWriter by design, allowing these to be end-to-end tests. Perhaps we should extract the `ChainReader` specific tests (i.e. ones that don't use `ChainWriter`) like [this one](https://github.com/smartcontractkit/chainlink-common/blob/b10e2708ab783ff269fac0d98a0dfc1624649c8c/pkg/types/interfacetests/chain_components_interface_tests.go#L82-L94) into their own test function. Similarly, once `ChainWriter` specific tests have been implemented, they can have their own test function. Then we have `ChainComponents` tests, which are these end-to-end tests I've created in this PR.

Reply:
These are all very distinct Chain Reading test cases , there is no functionality called `ChainComponentsGetLatestValue` its `ChainReaderGetLatestValue`. The purpose of these tests is for a developer that is doing a chain integration to be able to track the progress of implementing Chain Reader features. The testing of CW is a byproduct and we should never measure if CW works by these tests since CW should have its own tests that cover all of these use cases. This means that if any of these tests break it has to be obvious that its because of ChainReader not ChainWriter.

Reply:
> These are all very distinct Chain Reading test cases , there is no functionality called ChainComponentsGetLatestValue its ChainReaderGetLatestValue.

This is a very valid point. The main purpose of these tests are providing coverage to all `ChainReader` cases while also using an end-to-end flow via the `ChainWriter`. The only issue is that they are also testing `ChainWriter` inherently, making it inaccurate to say they are isolating the `ChainReader`'s functionality.

Reply:
I reverted the test names to be `ContractReader` specific, with the idea that we will add additional `ChainWriter` specific tests later.

Reply:
Talking to Silas about this earlier, I agree that it is possible that ChainWriter changes could break these changes, but I thought we were all okay with that to allow these tests to fully run E2E (so the Chain/Contract Reader actually reads data written by the Writer on chain). I think we should keep the tests functionally as they are in this PR, but revert the test names back to Contract/Chain Reader specific. In future PRs, we can add additional ChainWriter specific tests as we come up with ideas for what parts of the interface we want to specifically test.